test/MUXControlsTestApp/VisualTreeTestBase.cs

```diff
@@ -0,0 +1,360 @@
+            {
+                string fileName = String.Format("{0}-{1}.xml", fileNamePrefix, version);
+                if (MasterFileStorage.IsMasterFilePresent(fileName))
+                {
+                    return fileName;
+                }
+            }
+            {
+                string fileName = fileNamePrefix + ".xml";
+                if (MasterFileStorage.IsMasterFilePresent(fileName))
+                {
+                    return fileName;
+                }
+            }
+            return null;
+        }
+
+        public static string GetMasterFileContent(string fileName)
+        {
+            return GetMasterFileContentAsync(fileName).Result;
+        }
+
+        public static bool IsMasterFilePresent(string fileName)
+        {
+            try
+            {
+                GetMasterFileContentAsync(fileName).Wait();
+                return true;
+            }
+            catch (AggregateException ae)
+            {
+                ae.Handle((x) =>
+                {
+                    return (x is FileNotFoundException); // This we know how to handle.
+                });
+                return false;
+            }
+        }
+
+        public static void LogMasterFile(StorageFolder storageFolder, string fileName, string content)
+        {
+            LogMasterFileAsync(storageFolder, fileName, content).Wait();
+        }
+
+        private static readonly string MasterFileFullPathPrefix = "ms-appx:///master/";
+
+        private static async Task<string> GetMasterFileContentAsync(string fileName)
+        {
+            Uri uri = new Uri(MasterFileFullPathPrefix + fileName);
+            var file = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            return await FileIO.ReadTextAsync(file);
+        }
+
+        private static async Task LogMasterFileAsync(StorageFolder storageFolder, string fileName, string content)
+        {
+            var file = await storageFolder.CreateFileAsync(fileName, CreationCollisionOption.ReplaceExisting);
+            await FileIO.WriteTextAsync(file, content);
+        }
+    }
+
+    class VisualTreeOutputCompare
+    {
+        public VisualTreeOutputCompare(string xml1, string xml2)
+        {
+            var diffBuilder = new InlineDiffBuilder(new DiffPlex.Differ());
+            var diff = diffBuilder.BuildDiffModel(xml1, xml2);
+
+            foreach (var line in diff.Lines)
+            {
+                switch (line.Type)
+                {
+                    case ChangeType.Inserted:
+                        AddError("+ " + line.Text);
+                        break;
+                    case ChangeType.Deleted:
+                        AddError("- " + line.Text);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+        private StringBuilder _sb = new StringBuilder();
```

Comment:
This is only used for errors, right? Maybe call it _errors instead? StringBuilder is really an implementation detail.